pkg/commands/update/versioner.go

```diff
@@ -101,91 +81,83 @@ func (g GitHub) LatestVersion(ctx context.Context) (semver.Version, error) {
 func (g GitHub) Download(ctx context.Context, version semver.Version) (filename string, err error) {
 	releaseID, err := g.getReleaseID(ctx, version)
 	if err != nil {
-		return filename, err
+		return "", err
 	}
 
 	release, _, err := g.client.Repositories.GetRelease(ctx, g.org, g.repo, releaseID)
 	if err != nil {
-		return filename, fmt.Errorf("error fetching release: %w", err)
+		return "", fmt.Errorf("error fetching release: %w", err)
 	}
 
 	assetID, err := g.getAssetID(release.Assets)
 	if err != nil {
-		return filename, err
+		return "", err
 	}
 
 	rc, _, err := g.client.Repositories.DownloadReleaseAsset(ctx, g.org, g.repo, assetID, http.DefaultClient)
 	if err != nil {
-		return filename, err
+		return "", err
 	}
 	defer rc.Close()
 
-	var extension string
-
-	// TODO: We might need to also account for Window users by also checking for
-	// the .zip extension that goreleaser generates:
-	// https://github.com/fastly/cli/blob/26588cfd2d00d18643bac5cc18242b2d5ee84b34/.goreleaser.yml#L51
-	//
-	// Ideally the formats would be the same, but if that's not possible then
-	// we can look to use a genericised method such as
-	// https://pkg.go.dev/github.com/mholt/archiver#Extract for handling the
-	// extraction of a binary from the asset file instead of using the current
-	// archiver.NewTarGz().Extract() method.
-	if strings.HasSuffix(g.releaseAsset, ".tar.gz") {
-		extension = ".tar.gz"
+	dir, err := os.MkdirTemp("", "fastly-download")
+	if err != nil {
+		return "", fmt.Errorf("error creating release download directory: %w", err)
 	}
+	defer os.RemoveAll(dir)
 
-	tmp := os.TempDir()
-	dst, err := os.CreateTemp(tmp, fmt.Sprintf("%s_%s_*%s", g.binary, version, extension))
+	dst, err := os.Create(filepath.Join(dir, g.releaseAsset))
```

Comment:
Re-using the `dst` var name throughout the function is also creating unnecessary ambiguity, I think...

```suggestion
	assetFile, err := os.Create(filepath.Join(dir, g.releaseAsset))
```